vcd/datasource_vcd_org_vdc_test.go

```diff
@@ -0,0 +1,239 @@
+		}
+
+		if attr["compute_capacity.0.memory.0.used"] == vdcResource.Primary.Attributes["compute_capacity.0.memory.0.used"] {
+			return fmt.Errorf("compute_capacity.0.memory.0.used is %#v; want %#v", attr["compute_capacity.0.memory.0.used"], vdcResource.Primary.Attributes["compute_capacity.0.memory.0.used"])
+		}
+
+		if attr["storage_profile.0.name"] != vdcResource.Primary.Attributes["storage_profile.0.name"] {
+			return fmt.Errorf("storage_profile.0.name is %s; want %s", attr["storage_profile.0.name"], vdcResource.Primary.Attributes["storage_profile.0.name"])
+		}
+
+		if attr["storage_profile.0.limit"] != vdcResource.Primary.Attributes["storage_profile.0.limit"] {
+			return fmt.Errorf("storage_profile.0.limit is %s; want %s", attr["storage_profile.0.limit"], vdcResource.Primary.Attributes["storage_profile.0.limit"])
+		}
+
+		if attr["storage_profile.0.enabled"] != vdcResource.Primary.Attributes["storage_profile.0.enabled"] {
+			return fmt.Errorf("storage_profile.0.enabled is %s; want %s", attr["storage_profile.0.enabled"], vdcResource.Primary.Attributes["storage_profile.0.enabled"])
+		}
+
+		if attr["storage_profile.0.default"] != vdcResource.Primary.Attributes["storage_profile.0.default"] {
+			return fmt.Errorf("storage_profile.0.default is %s; want %s", attr["storage_profile.0.default"], vdcResource.Primary.Attributes["storage_profile.0.default"])
+		}
+
+		if attr["metadata.vdc_metadata"] != vdcResource.Primary.Attributes["metadata.vdc_metadata"] {
+			return fmt.Errorf("metadata.vdc_metadata is %s; want %s", attr["metadata.vdc_metadata"], vdcResource.Primary.Attributes["metadata.vdc_metadata"])
+		}
+
+		return nil
+	}
+}
+
+func importStateIdByVdc(objectName string) resource.ImportStateIdFunc {
+	return func(*terraform.State) (string, error) {
+		importId := testConfig.VCD.Org + "." + objectName
+		if testConfig.VCD.Org == "" || objectName == "" {
+			return "", fmt.Errorf("missing information to generate import path: %s", importId)
+		}
+		return importId, nil
+	}
+}
+
+const testAccCheckVcdVdcDatasource_basic = `
+resource "vcd_org_vdc" "{{.VdcName}}" {
+  name = "{{.VdcName}}"
+  org  = "{{.OrgName}}"
+
+  allocation_model = "{{.AllocationModel}}"
+  network_pool_name     = "{{.NetworkPool}}"
+  provider_vdc_name     = "{{.ProviderVdc}}"
+
+  compute_capacity {
+    cpu {
+      allocated = 2048
+      limit     = 2048
+    }
+
+    memory {
+      allocated = 2048
+      limit     = 2048
+    }
+  }
+
+  storage_profile {
+    name = "{{.ProviderVdcStorageProfile}}"
+    enabled  = true
+    limit    = 10240
+    default  = true
+  }
+
+  metadata = {
+    vdc_metadata = "VDC Metadata"
+  }
+
+  enabled                  = true
+  enable_thin_provisioning = true
+  enable_fast_provisioning = true
+  delete_force             = true
+  delete_recursive         = true
+}
+
+data "vcd_org_vdc" "asDatasource"{
```

Comment:
This is a hackish way of using a data source. We can use a data source for an existing resource, but we shouldn't.
The test should use a data source for an existing entity, without defining a resource for it.

Reply:
But probably depends if we have it. If we don't it is cheaper to set it up such way (although in practice it wouldn't be useful in any way)

Reply:
Yes it is problem `if we have it`, so I create to have it.

Reply:
Our test suite requires a VDC. Thus we have it, and we can make a data source for it. Similar for what we do for Org and catalog.

Reply:
Yes, that's true, but in that way I have to expect everywhere it will be same and hardcode values which I don't have control.

Reply:
Please have a look at `TestAccVcdDatasourceOrg` or `TestAccVcdDNAT_WithExtNetw`.
There is no need for either hard coding or knowing all the values in advance.

Reply:
Changed accordingly